app/services/nomis/api.rb

```diff
@@ -4,33 +4,31 @@ module Nomis
   NotFound = Class.new(Error)
 
   class Api
-    class << self
-      def enabled?
-        Rails.configuration.nomis_api_host != nil
-      end
+    include Singleton
 
-      def instance
-        unless enabled?
-          fail DisabledError, 'Nomis API is disabled'
-        end
+    def self.enabled?
+      Rails.configuration.nomis_api_host != nil
+    end
 
-        @instance ||= begin
-          host = Rails.configuration.nomis_api_host
-          client_token = Rails.configuration.nomis_api_token
-          client_key = Rails.configuration.nomis_api_key
-          client = Nomis::Client.new(host, client_token, client_key)
-          new(client)
-        end
+    def initialize
+      unless self.class.enabled?
+        fail DisabledError, 'Nomis API is disabled'
       end
-    end
 
-    def initialize(client)
-      @client = client
+      pool_size = Rails.configuration.connection_pool_size
+      @pool = ConnectionPool.new(size: pool_size, timeout: 1) do
```

Comment:
Probably want the timeout  to be configurable as ENV var so it can easily be tweaked when needed (i.e a new release when the NOMIS API is slower than usual)